pkg/profiler/profile_writer.go

```diff
@@ -58,18 +59,26 @@ func (fw *FileProfileWriter) Write(_ context.Context, labels model.LabelSet, pro
 // RemoteProfileWriter is a profile writer that writes profiles to a remote profile store.
 type RemoteProfileWriter struct {
 	profileStoreClient profilestorepb.ProfileStoreServiceClient
+	pool               sync.Pool
 }
 
 // NewRemoteProfileWriter creates a new RemoteProfileWriter.
 func NewRemoteProfileWriter(profileStoreClient profilestorepb.ProfileStoreServiceClient) *RemoteProfileWriter {
 	return &RemoteProfileWriter{
 		profileStoreClient: profileStoreClient,
+		pool: sync.Pool{New: func() interface{} {
```

Comment:
Also, we have a wrapper around the remote writer that batches the requests. Let's make sure they play nicely.

Reply:
ooooo good point! That would probably cause a race like this.

Reply:
You're right guys, `func (b *BatchWriteClient) WriteRaw()` returns a response and sends the series async, so the byte slice might be already unusable https://github.com/parca-dev/parca-agent/blob/main/pkg/agent/batch_remote_write_client.go#L183.